byron/ledger/executable-spec/test/Generator.hs

```diff
@@ -0,0 +1,142 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE TupleSections       #-}
+
+module Generator where
+
+import           Data.Bits (shiftL)
+import           Data.Word (Word64)
+import           Hedgehog
+import qualified Hedgehog.Gen    as Gen
+import qualified Hedgehog.Range  as Range
+import           Ledger.Core (Slot(..), SlotCount(..), BlockCount(..))
+import           Ledger.GlobalParams (k)
+import           Ledger.Update (ProtVer(..), PParams(..))
+import           Numeric.Natural (Natural)
+
+-- | Generates a slot within the given bound
+slotGen :: Word64 -> Word64 -> Gen Slot
+slotGen lower upper = do
+  Slot <$> Gen.word64 (Range.linear lower upper)
+
+-- | Generates a 'ProtVer'
+protVerGen :: Gen ProtVer
+protVerGen =
+  (\(a, b) alt -> ProtVer a b alt) <$>
+    minMaj <*>
+    Gen.integral (Range.linear (0 :: Natural) 100)
+ where
+  minMaj :: Gen (Natural, Natural)
+  minMaj = do
+    pvMin  <- Gen.integral (Range.linear (0 :: Natural) 100)
+    pvDiff <- Gen.integral (Range.linear (0 :: Natural) 20)
+    pure (pvMin, pvMin + pvDiff)
```

Comment:
I don't quite understand what this is doing - what's the relationship being captured here?

Reply:
Oh, this was completely off. It is fixed now. Thanks for spotting it!